helix-rest/src/main/java/org/apache/helix/rest/metadatastore/ZkMetadataStoreDirectory.java

```diff
@@ -322,14 +322,19 @@ public void refreshRoutingData(String namespace) {
     }
 
     Map<String, List<String>> rawRoutingData;
+    // Remove the raw data first in case of failure on creation
+    _realmToShardingKeysMap.remove(namespace);
     try {
       rawRoutingData = _routingDataReaderMap.get(namespace).getRoutingData();
       _realmToShardingKeysMap.put(namespace, rawRoutingData);
     } catch (InvalidRoutingDataException e) {
+      _routingDataMap.remove(namespace);
       LOG.error("Failed to refresh cached routing data for namespace {}", namespace, e);
       return;
     }
 
+    // Remove routing data first in case of failure on creation
+    _routingDataMap.remove(namespace);
```

Comment:
Why don't you just put `_routingDataMap.remove(namespace);` up there next to `_realmToShardingKeysMap.remove(namespace);`?

Why is there a need for this duplicate code? routingDataMap entry should be cleared anyway prior to update, right?

Reply:
Makes sense, let me change that.